odbc/src/api/functions.rs

```diff
@@ -1114,24 +1116,69 @@ pub unsafe extern "C" fn SQLGetConnectAttr(
 /// This is the WChar version of the SQLGetConnectAttr function
 ///
 /// # Safety
-/// Because this is a C-infereface, this is necessarily unsafe
+/// Because this is a C-interface, this is necessarily unsafe
 ///
 #[no_mangle]
 pub unsafe extern "C" fn SQLGetConnectAttrW(
-    _connection_handle: HDbc,
-    _attribute: ConnectionAttribute,
-    _value_ptr: Pointer,
-    _buffer_length: Integer,
-    _string_length_ptr: *mut Integer,
+    connection_handle: HDbc,
+    attribute: ConnectionAttribute,
+    value_ptr: Pointer,
+    buffer_length: Integer,
+    string_length_ptr: *mut Integer,
 ) -> SqlReturn {
-    unimplemented!()
+    let mut err = None;
+    let conn_handle = MongoHandleRef::from(connection_handle);
+
+    // This scope is introduced to make the RWLock Guard expire before we write
+    // any error values via add_diag_info as RWLock::write is not reentrant on
+    // all operating systems, and the docs say it can panic.
+    let sql_return = {
+        let conn = must_be_valid!((*conn_handle).as_connection());
+        let attributes = &conn.read().unwrap().attributes;
+
+        match attribute {
+            ConnectionAttribute::CurrentCatalog => {
+                let current_catalog = attributes.current_catalog.as_ref().map(|s| s.as_str());
+                match current_catalog {
+                    None => SqlReturn::NO_DATA,
+                    Some(cc) => set_output_string(
+                        cc,
+                        value_ptr as *mut WChar,
+                        buffer_length as usize,
+                        string_length_ptr as *mut SmallInt,
+                    ),
+                }
+            }
+            ConnectionAttribute::LoginTimeout => {
+                let login_timeout = attributes.login_timeout.unwrap_or(0);
+                set_output_fixed_data(&login_timeout, value_ptr, string_length_ptr as *mut Len)
```

Comment:
Good point. I'm not sure what we should do here since the spec doesn't say what to do if the value_ptr is null and the value is not a string. What do you think we should do in this case?

Reply:
The trick to know what the DM, is to try it out 😅 You don't have to use our driver though, any ODBC driver would work, but you need to install one and call the ODBC API directly, you can't use unit test for this.
Otherwise, yes, the spec is not clear, so we should be able to pick whichever solution, since no one can tell us that we are "wrong" 😜

Reply:
This goes away with the new code too anyway 🎉